### PR TITLE
Separate tribes

### PR DESCRIPTION
**Pull request by gunchleoc**
Separate the definitions of each tribe into separate files

* Get rid of global init and preload files
* Define each tribe separately in a `tribes/<tribe_name>` directory
* Shift starting condition and image files into each tribe's directory
* Shift wares' and workers' `default_target_quantity` and `preciousness` to tribe definitions
* Reduce number of disk reading operations in game loading screens

There will be another branch for the tribe-dependent help files.

### CODE REVIEW

**Review by hessenfarmer — APPROVED** [review]
Code LGTM.
Not tested.
only the tribes/tribes Pathname looks confusing, what about tribes/basics/ or tribes/basic_info or tribes/initialisation

**Comment by hessenfarmer** [comment]
BTW: I hope you are aware that all the refactoring does cost me some time to adopt amazons as well ;-)

**Comment by gunchleoc** [comment]
> BTW: I hope you are aware that all the refactoring does cost me some time to adopt amazons as well ;-)

Yes, I am. We need this for moddability though. This series of branches is the only one you need to implement right now, the program refactorings all have compatibility code and can be done later.

Do let me know if you want my help, I'd be happy to do the conversion for the amazons.

> what about tribes/basics/ or tribes/basic_info or tribes/initialisation

When I refactor the world too, we might end up with something different, e.g. `units/tribes` and `units/world` or something like that. I am planning to refactor the world too in order to keep our API orthogonal.

**Comment by hessenfarmer** [comment]
> 
> 
> > BTW: I hope you are aware that all the refactoring does cost me some time to adopt amazons as well ;-)
> 
> Yes, I am. We need this for moddability though. This series of branches is the only one you need to implement right now, the program refactorings all have compatibility code and can be done later.
> 
> Do let me know if you want my help, I'd be happy to do the conversion for the amazons.    
    
Well I have already done most of it and just need to incorporate the tribes_filtering branch and this one. At least i do learn and understand a lot of things while doing so. So no worries. Better moddability is worth the pain in any case.  

> 
> > what about tribes/basics/ or tribes/basic_info or tribes/initialisation
> 
> When I refactor the world too, we might end up with something different, e.g. `units/tribes` and `units/world` or something like that. I am planning to refactor the world too in order to keep our API orthogonal.

This I don't understand. I think the API's should be as consistnet as they can be. However we should not mix the world and the tribes in their file structure. So Ifound it just confusing and tried to propose a more "telling" folder name.

**Comment by gunchleoc** [comment]
I'll definitely keep the world and tribes separate for now. Merging paths at the end can always be done more easily than pulling them apart again. It will be easier to make this decision once everything else has been done.

Let's go with `tribes/initialisation` for now.

File: doc/sphinx/extract_rst.py [hunk 1]
```diff
@@ -38,7 +38,7 @@
     ('data/scripting', '', 'auxiliary'),
     ('data/scripting/win_conditions', '', 'auxiliary'),
     ('data/scripting/editor', '', 'lua_world_other'),
-    ('data/tribes/tribes/atlanteans', '', 'lua_tribes_defining'),
+    ('data/tribes/initialisation/atlanteans', '', 'lua_tribes_defining'),
```
Comment (hessenfarmer) [inline doc/sphinx/extract_rst.py]: now we have a typo here I believe

File: utils/buildcat.py [hunk 2]
```diff
@@ -69,8 +69,7 @@
         '../../data/world/*/*/*/*/*/*.lua',
     ]),
     ('tribes/tribes', [
-        '../../data/tribes/tribes/*/menu.lua',
-        '../../data/tribes/tribes/*/starting_conditions/*.lua',
+        '../../data/tribes/initialisation/*/starting_conditions/*.lua',
```
Comment (hessenfarmer) [inline utils/buildcat.py]: typo

File: data/tribes/initialization/atlanteans/units.lua [hunk 3]
```diff
@@ -32,7 +32,7 @@
 -- Configuring a Tribe's Units Usage
 -- ---------------------------------
 --
--- The ``tribes/tribes/<tribename>/units.lua`` file configures the units that a tribe can use.
+-- The ``tribes/initialisation/<tribename>/units.lua`` file configures the units that a tribe can use.
```
Comment (hessenfarmer) [inline data/tribes/initialization/atlanteans/units.lua]: typo initialisation --> initialization

File: data/tribes/initialization/atlanteans/units.lua [hunk 4]
```diff
@@ -9,7 +9,7 @@
 --
 -- Each tribe needs to have some basic information defined for displaying it in menus
 -- and for loading its game starting conditions.
--- The file **has to be located in** ``tribes/tribes/<tribename>/init.lua`` so that the engine can find it.
+-- The file **has to be located in** ``tribes/initialisation/<tribename>/init.lua`` so that the engine can find it.
```
Comment (hessenfarmer) [inline data/tribes/initialization/atlanteans/units.lua]: typo

File: utils/buildcat.py [hunk 5]
```diff
@@ -69,8 +69,7 @@
         '../../data/world/*/*/*/*/*/*.lua',
     ]),
     ('tribes/tribes', [
```
Comment (hessenfarmer) [inline utils/buildcat.py:71]: does this need to be changed as well?

**Review by hessenfarmer — CHANGES_REQUESTED** [review]
found some typo to be corrected

**Comment by gunchleoc** [comment]
Should be all fixed now. I went with `tribes/initialization` in the end, since we're using American and not British English.

**Comment by hessenfarmer** [comment]
> 
> 
> Should be all fixed now. I went with `tribes/initialization` in the end, since we're using American and not British English.

that's why I just commented the BE String.  What about the last unresolved comment though?

**Comment by gunchleoc** [comment]
> > Should be all fixed now. I went with `tribes/initialization` in the end, since we're using American and not British English.
> 
> that's why I just commented the BE String. What about the last unresolved comment though?

I don't know what I'm supposed to do there, I have already fixed the path for the starting conditions.

**Review by hessenfarmer — APPROVED** [review]
forget my last comment I was confused by only wathing the diff tribes/tribes is not a pathname there.
Code LGTM. 
not tested

**Comment by hessenfarmer** [comment]
while adopting amazons it arose to me the question what might be the reason of only atlanteans having the toolbar images?

**Comment by gunchleoc** [comment]
> while adopting amazons it arose to me the question what might be the reason of only atlanteans having the toolbar images?

They are placeholder image to show that the code works. We need an art contribution to give the toolbar a nice frame. https://github.com/widelands/widelands/issues/1092

I also kept them optional so you don't have to provide any, lacking an artist.

**Pull request merged**